lib/hocon/impl/config_reference.rb

```diff
@@ -72,4 +73,56 @@ def resolve_substitutions(context, source)
     end
 
   end
+
+  def self.not_resolved
```

Comment:
This method is private in the java version.  I've been trying to keep that consistent in the ruby code.  There are probably instances of that in here ... might not be worth trying to fix 'em all now, but food for thought as we continue down this long and winding road ... err, iceberg.  Long and winding iceberg.